tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp

```diff
@@ -116,6 +116,32 @@ bool RemoveBufferBlockVisitor::updateStorageClass(
     }
   }
 
+  // For pointer-to-struct cases, we may need to update the storage class for
+  // the inner struct fields.
+  if (const auto *innerStructType = dyn_cast<StructType>(innerType)) {
+    auto fields = innerStructType->getFields();
+    std::vector<StructType::FieldInfo> newFields(fields.begin(), fields.end());
+    for (size_t i = 0; i < fields.size(); i++) {
+      if (const auto *innerPtrType =
+              dyn_cast<SpirvPointerType>(fields[i].type)) {
+        if (hasStorageBufferInterfaceType(innerPtrType->getPointeeType()) &&
+            innerPtrType->getStorageClass() !=
+                spv::StorageClass::StorageBuffer) {
+          auto *newInnerType = context.getPointerType(
+              innerPtrType->getPointeeType(), spv::StorageClass::StorageBuffer);
+          newFields[i] = newInnerType;
+          *newType = context.getPointerType(
```

Comment:
When there are multiple such members in the struct, we have to handle them as well.
Could you please move lines 133~139 out of this for-loop to handle multiple members (I mean fields)?

Reply:
Oops, that was why I introduced a separate newFields and then apparently didn't finish the fix. Fixed now, thanks!